engine/src/cmd/basecomputer.cpp

```diff
@@ -114,8 +116,8 @@ std::vector<std::string> getWeapFilterVec() {
 std::vector<std::string> weapfiltervec = getWeapFilterVec();
 
 bool upgradeNotAddedToCargo(std::string category) {
-    for (unsigned int i = 0; i < weapfiltervec.size(); ++i) {
-        if (weapfiltervec[i].find(category) == 0) {
+    for (const auto & i : weapfiltervec) {
```

Comment:
I personally don't have a problem with auto, but @BenjamenMeyer frowns upon them. 
Therefore, where I can deduce the type, I'll use it.

Reply:
@BenjamenMeyer has said previously that he's OK with auto for iterators.